packages/hash/integration/src/tests/model/knowledge/entity.model.test.ts

```diff
@@ -26,81 +27,101 @@ const graphApi = createGraphClient(logger, {
-    accountId,
-    schema: {
-      kind: "dataType",
-      title: "Text",
-      type: "string",
-    },
-  }).catch((err) => {
-    logger.error(`Something went wrong making Text: ${err}`);
-    throw err;
-  });
+describe("Entity CRU", () => {
+  let accountId: string;
+  let entityTypeModel: EntityTypeModel;
+  let textDataTypeModel: DataTypeModel;
+  let namePropertyTypeModel: PropertyTypeModel;
+  let favoriteBookPropertyTypeModel: PropertyTypeModel;
+  let linkTypeFriend: LinkTypeModel;
+
+  beforeAll(async () => {
+    const testUser = await createTestUser(graphApi, "entitytest", logger);
 
-  const results = await Promise.all([
-    EntityTypeModel.create(graphApi, {
+    accountId = testUser.entityId;
+
+    textDataTypeModel = await DataTypeModel.create(graphApi, {
       accountId,
       schema: {
-        kind: "entityType",
-        title: "Person",
-        pluralTitle: "People",
-        type: "object",
-        properties: {},
+        kind: "dataType",
+        title: "Text",
+        type: "string",
       },
     }).catch((err) => {
-      logger.error(`Something went wrong making Person: ${err}`);
+      logger.error(`Something went wrong making Text: ${err}`);
       throw err;
-    }),
-    PropertyTypeModel.create(graphApi, {
-      accountId,
-      schema: {
-        kind: "propertyType",
-        title: "Favorite Book",
-        pluralTitle: "Favorite Books",
-        oneOf: [{ $ref: textDataTypeModel.schema.$id }],
-      },
-    })
-      .then((val) => {
-        favoriteBookPropertyTypeModel = val;
+    });
+
+    await Promise.all([
+      EntityTypeModel.create(graphApi, {
+        accountId,
+        schema: {
+          kind: "entityType",
+          title: "Person",
+          pluralTitle: "People",
+          type: "object",
+          properties: {},
+        },
       })
-      .catch((err) => {
-        logger.error(`Something went wrong making Favorite Book: ${err}`);
-        throw err;
-      }),
-    PropertyTypeModel.create(graphApi, {
-      accountId,
-      schema: {
-        kind: "propertyType",
-        title: "Name",
-        pluralTitle: "Names",
-        oneOf: [{ $ref: textDataTypeModel.schema.$id }],
-      },
-    })
-      .then((val) => {
-        namePropertyTypeModel = val;
+        .then((val) => {
+          entityTypeModel = val;
+        })
+        .catch((err) => {
+          logger.error(`Something went wrong making Person: ${err}`);
+          throw err;
+        }),
+      LinkTypeModel.create(graphApi, {
+        accountId,
+        schema: {
+          kind: "linkType",
+          title: "Friends",
+          pluralTitle: "Friends",
+          description: "Friend of",
+        },
       })
-      .catch((err) => {
-        logger.error(`Something went wrong making Names: ${err}`);
-        throw err;
-      }),
-  ]);
-
-  entityTypeModel = results[0];
-});
+        .then((val) => {
+          linkTypeFriend = val;
+        })
```

Comment:
Nit: you don't need these assignments if you make use of the returned array of `new Proimse(...)` like this:


```TypeScript
[entityTypeModel, favoriteBookPropertyTypeModel, namePropertyTypeModel] =
    await Promise.all([ ... ]);
```

Reply:
You wouldn't believe how time I spent some weeks ago trying to figure out this syntax (without success).
Thanks so much for pointing this out, I really wanted this when I did index-assignment in these tests 4dffc2c3f794ba1d021695c66de56656cc58b0f7 :smile:

Reply:
Interesting. I actually went away from this syntax the other day because this is really vulnerable to re-ordering bugs.

Reply:
No worries, I was surprised this worked when I first tried it - but it's grown on me. It is a very JS way of going about things though, but I think that's okay.

Reply:
That's a good point. What's the stance on this?

Reply:
I don't feel super strongly about this but I think I'd prefer the older syntax. Especially considering we're already having to be pretty verbose with individual handlers (the `.catch` statements and such), I don't think we gain much by doing this. But I'm open to it

Reply:
The re-ordering bugs would only happen when reordering entries of the same type i suppose

Reply:
I think the problem with re-ordering might lead to some annoyances, so perhaps we should keep that.

Reply:
Reverted in 51a3f19cf74561ab86854d7a5870e5a6130537c1
I think the main argument for using the callback instead of array binding for this model class is that we would likely touch the entity tests a lot.